controllers/trivy_vuln_crds_controller.go

```diff
@@ -0,0 +1,219 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strconv"
+)
+
+const LagoonProjectLabel = "lagoon.sh/project"
+const TrivyVulnReportNamespace = "trivy-operator.resource.namespace"
+
+type TrivyVulnerabilityReportReconciler struct {
+	Client         client.Client
+	Scheme         *runtime.Scheme
+	MessageQWriter func(data []byte) error
+	WriteToQueue   bool
+	PredicateData  interface{}
+}
+
+type ProblemsPayload struct {
+	Problems []interface{} `json:"problems"`
+}
+
+// predicate that filters out non-lagoon projects
+func (r *TrivyVulnerabilityReportReconciler) lagoonProjectLabelsOnlyPredicate() predicate.Predicate {
+
+	return predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			if r.isLagoonProject(event.Object) &&
+				!insightsProcessedAnnotationExists(event.Object) {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			if r.isLagoonProject(event.ObjectNew) &&
+				!insightsProcessedAnnotationExists(event.ObjectNew) {
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			return false
+		},
+	}
+}
+
+func (r *TrivyVulnerabilityReportReconciler) isLagoonProject(event client.Object) bool {
```

Comment:
This function probably shouldn't be a method of the TrivyVulnerabilityReportReconciler. It suggests, to me, that we're asking if the TrivyVulnerabilityReportReconciler is _itself_ a lagoon project when you read it in code. This is a more generic function - I'd just have a function `isLagoonProject` where you can pass in, say,  a namespace object and it returns a true/false.